### ISSUE TITLE
[Request] Create Eloquent Builder for Model

### ISSUE THREAD

**Issue by Marko298** [laravel-idea/plugin#81]
Is it possible to add generation of Eloquent Builder for Model and add ```newEloquentBuilder``` with this builder?

**Comment by adelf** [comment]
Hi, Mark.

You mean class depending on Eloquent/Builder class and Eloquent can use it instead of standard one? Could you provide an example?

**Comment by Marko298** [comment]
You are correct

Here is example model
```php
<?php

namespace App\Models;

use App\Builders\AuthorBuilder;
use App\Models\Collections\AuthorCollection;
use Illuminate\Database\Eloquent\Model;

class Author extends Model
{
    public function newCollection(array $models = [])
    {
        return new AuthorCollection($models);
    }

    public function newEloquentBuilder($query)
    {
        return new AuthorBuilder($query);
    }
}

```

Example builder
```php
<?php

namespace App\Builders;

use Illuminate\Database\Eloquent\Builder;

class AuthorBuilder extends Builder
{
    public function nameStartsWith(string $letter)
    {
        return $this->whereRaw('SOME SQL', $letter);
    }
}



```

**Comment by Marko298** [comment]
it provides better autocomplete if you write queries like
``` Author::query()->nameStartsWith('A')```

**Comment by adelf** [comment]
Thanks. I’ll try to implement it and use these builder and collection classes in my Eloquent Helper Code generation.

**Comment by Marko298** [comment]
Thanks a lot! Love your work, it saves some time and reduces amount of errors

**Comment by adelf** [comment]
It's implemented and will be included in the next version.

**Comment by adelf** [comment]
Implemented in Laravel Idea 3.4. Please check how it works and report here if something is wrong.
